0000-server-roles.md

```diff
@@ -0,0 +1,71 @@
+
+All side protocols are considered to be ledger-specific, for instance see https://github.com/interledgerjs/ilp-plugin-xrp-asym-server/issues/10 in the case of XRP payment channels.
+
+### BTP/2.0
+BTP/2.0 is defined in [BTP (draft 7)](https://interledger.org/rfcs/0023-bilateral-transfer-protocol/draft-7.html).
+
+### BTP/2.0-I
+BTP/2.0-I is defined as BTP/2.0 with idempotency: each request should be repeated regularly until a response is received, and each time a repeated request is received, the same response should be given. This means each peer needs to remember the responses it send until the sender has stopped repeating that request, and has moved on to newer requests.
+
+### HTTP-OER-ASYNC
+HTTP-OER-ASYNC is defined as follows:
+* each peer can make https POST requests to a webserver controlled by the other peer, using a bearer token in an Authorization header
+* the URL path starts with a `/`, followed by the protocolName (`'ilp'` for Interledger packets, other protocolNames like `'paychan'` can be used for bilateral settlement)
+* for protocolName `'ilp'`:
+  * the path continues with a second `/`, and a unique request identifier string, so the URL looks like `https://example.com/ilp/3442196b-5fc7-4d60-a17e-a9416a77745b`.
+  * the request body is an Interledger packet.
+  * the response status code is 200 if the packet was received correctly
+  * requests will be repeated idempotently until a 200 response is received
+  * Fulfill and Reject requests use the same request identifier string as the Prepare request they correspond to
+
+### HTTP-OER-SYNC
+HTTP-OER-SYNC is like HTTP-OER-ASYNC, except that the Fulfill or Reject packet is immediately included in the response body of the Prepare request.
+This means that there is no way to know whether Fulfill or Reject packets were delivered successfully, except for the assumption that if they were
+not delivered successfully, then the other party would repeat the request.
+
+## Connector requirements
+On top of these bilateral relations, apart from the side protocols, the `'ilp'` protocol carries ILP packets for various purposes.
+The request has the form of an ILP Prepare, and the response has the form of an ILP Fulfill (if successful) or an ILP Reject (if unsuccessful).
+Encoded ILP-packets, the request types a connector should respond to are: ilp/forwarded, ilp/il-dcp, ilp/il-balance, and route-broadcast.
+
+#### forwarded
+If the destination address does not start with `peer.`, then the connector should try to obtain the fulfillment by forwarding the payment to the destination address. It may charge a reasonable transaction fee, require a reasonable amount of time to pass on prepares and pass back fulfills, and have a reasonable failure rate.
```

Comment:
s/forwarding the payment to the destination address/forwarding the ILP packet to the destination address/